### ISSUE TITLE
Make updates on tag component

### ISSUE THREAD

**Issue by hazalarpalikli** [alphagov/govuk-frontend#6637]
## What
Update the Tag component with the new design decisions

## Why
So that the tag component is accessible

## Who needs to work on this
Developer

## Who needs to review this
Designer 

## Done when

- [x] Tag border is removed 
- [x] Yellow tag background and text is updated 
- [x] Grey tag background and text is updated 
- [x] Light blue tag is retired 
- [x] Changelog updated
- [x] Release notes updated

**Comment by 36degrees** [comment]
Thinking about the release notes, I think we need to cover the removal of the `--light-blue` modifier class.

Otherwise, should the wider colour changes be rolled up into the section about the wider palette changes?

(We should include the border removal in the release notes for the release candidate, but I think as it was added and then removed within v6 betas we don't need to include it in the final release notes)

**Comment by hazalarpalikli** [comment]
List of things that needs updating 

1. Remove all the border from all of the tags 
2. Retire light blue tag 

Yellow tag:
1. Change the text colour to: `#7A3C1C` / Orange shade 50%
2. Change the background colour to: `#FFEE80`/ Yellow tint 50% 

Grey tag:
1. Change the text colour to: `#0B0C0C` / Black 
2. Change the background colour to: `#CECECE` /  Black tint 80%

All the accessibility checks are done on the yellow and grey tag. Results are below:

| Tag | Text colour | Background colour | Font size | Contrast ratio | WCAG AA | WCAG AAA | APCA value | APCA font note | APCA
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
| <img width="121" height="35" alt="yellow tag" src="https://github.com/user-attachments/assets/7d669e41-6769-4568-b55a-06455df0c3ed" /> | #7A3C1C | #FFEE80 | 19 | 7.14 | Pass | Pass | 78.3 Lc | Standard (Body) | Pass
| <img width="120" height="34" alt="Grey tag" src="https://github.com/user-attachments/assets/ecda0b9f-b411-4aa6-bcae-e1c54a674843" />  | #0B0C0C | #CECECE | 19 | 12.45 | Pass | Pass | 81.7 Lc | Standard (Body) | Pass

**Comment by Izabela-16** [comment]
Concern: Tags had become too tall, causing slight text overlap.
Brett pushed a change to revert the problematic change.
Next step: A designer @hazalarpalikli is needed to look at it. Hazal tagged on the PR.
